functions/_nvm_use_on_pwd_change.fish

```diff
@@ -0,0 +1,9 @@
+function _nvm_use_on_pwd_change \
```

Comment:
We only need to define the PWD change function when the feature flag is set. This may involve defining another function inside the function or some other trick. 🤔

Reply:
I was thinking about using `funcsave` but the manual states:
> Note that because fish loads functions on-demand, **saved functions will not function as [event handlers](https://fishshell.com/docs/current/language.html#event) until they are run or sourced otherwise**. To activate an event handler for every new shell, add the function to your [configuration file](https://fishshell.com/docs/current/index.html#configuration) instead of using funcsave.

So need to add a call in `conf.d/nvm.fish`

Reply:
Could we get even fancier? Consider: 

```fish
function _nvm_use_on_pwd_change --argument-names _ op --on-variable nvm_use_on_pwd_change
    if test "$op" = ERASE
        functions --erase _nvm_pwd_change
    else
        function _nvm_pwd_change --on-variable PWD
            nvm use --silent 2>/dev/null
        end
    end
end
```